### ISSUE TITLE
Questions about the paper.

### ISSUE THREAD

**Issue by XuyangBai** [zzhang1987/Deep-Graphical-Feature-Learning#1]
Hi @zzhang1987 . Thanks for your sharing. I am doing the 2D image feature matching and 3D point cloud feature matching and I find your paper related and quite interesting. Here I have some questions regarding the paper.  

Since you are using the PointNet as a feature extractor, have you tried some advanced network architecture designed to replace PointNet such as [DGCNN](https://arxiv.org/abs/1801.07829), [KPConv](https://arxiv.org/abs/1904.08889) or [DCP](https://arxiv.org/abs/1905.03304) (which also deal with feature matching problem)? Actually I think most of the work on point cloud feature learning can be used in your case and they are also only using the point coordinate as input. Is there any specific reason why you are not comparing with these methods?

My second question is for experiment on PF-Pascal (sorry I am not familiar with this dataset so my understanding might be wrong), each image pair only have 6-18 manually annotated ground truth correspondence, then how large the `k` should be? In this situation, will the k-nn neighbor be too far away from the center point?

Thank you in advance!

**Comment by zzhang1987** [comment]
In fact I did not use PointNet as the feature extractor. I am using Message Passing Neural Network (MPNN) published in ICML 2017, which can be viewed as a general graph neural network framework ( in fact various graph neural networks including DGCNN can be viewed as a special case of MPNN). DCP is also published in ICCV19, thus I can not compare with it before its publication. Also DCP is designed for 3D rigid cases, for 2d graph matching, it may require additional adjustment.

PointNet is one of the most representative work of processing point clouds using neural network, and the authors showed that it can be used for matching in its supplementary file. Thus I choose to compare with PointNet.  Also DGCNN and KPConv are mainly proposed for segmentation and classification rather than matching and thus they are not compared. 

For PF-Pascal, I am using k=4. The k-nn neighbor can be far away from the center point. However, this happens in both reference and target graph, and thus the network can still extract enough information from structure similarities.

**Comment by XuyangBai** [comment]
Thank you for the explanation.  

I agree that DGCNN and KPConv are mainly proposed for segmentation and classification, but before the classification head and segmentation head, they are also trying to transform coordinates of feature points into local features, which is the same with your network. So I think these methods might also be suitable for the matching problem if trained with proper loss formulation. But anyway they are focusing on different tasks. 

Thank you again for your time.
